pool.go

```diff
@@ -44,7 +44,7 @@ type (
 	Process interface {
 		// Start runs the process. It returns an error object if any thing wrong
 		// happens in runtime.
-		Start() error
+		Start(ctx context.Context) error
```

Comment:
suggestion: we can have two `start` functions:
- Start()
- StartWithContext(ctx context.Context)